### PR TITLE
Feature fix profiles

### PR DESCRIPTION
**Pull request by Rene2mt**
# Committer Notes

This PR fixes the FedRAMP profile issues that were causing the resolved profile catalogs to fail validation, including:

- ensuring part ids are unique (no duplicates)
- adding namespace on parts where required
- fix issues with some of the parameters (issues #558) 
- consolidate FedRAMP's control alterations to one profile 
- simplify high moderate, and low profile to import new consolidated alterations profile
  - override alterations in cases where there are minor differences (e.g., between high, moderate, and low FedRAMP Additional Guidance / Requirements)

### All Submissions:

- [x] Have you selected the correct base branch per [Contributing](https://github.com/GSA/fedramp-automation/blob/master/CONTRIBUTING.md) guidance?
- [ ] Have you set "[Allow edits and access to secrets by maintainers](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork)"?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/GSA/fedramp-automation/pulls) for the same update/change?
- [x] Have you squashed any non-relevant commits and commit messages? \[[instructions](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History)\]
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] If applicable, have all [FedRAMP Documents Related to OSCAL Adoption](https://github.com/GSA/fedramp-automation) affected by the changes in this issue have been updated.?
- [ ] If applicable, does this PR reference the issue it addresses and explain how it addresses the issue?

By submitting a pull request, you are agreeing to provide this contribution under the [CC0 1.0 Universal public domain](https://creativecommons.org/publicdomain/zero/1.0/) dedication.

### CODE REVIEW

**Review by aj-stein-gsa — COMMENTED** [review]
@Rene2mt could you rebase when you are available and quickly confirm with a yes/no/maybe, the answer is more complicated answer to this question:

I am working on fixing up some intermittent CI/CD failures, in this GHA workflow failure on catalog and resolved catalog fixed by this work? Can we rebase?

https://github.com/GSA/fedramp-automation/actions/runs/10783460864/job/29905416403?pr=667

**Comment by aj-stein-gsa** [comment]
So @Rene2mt and @brian-ruf, we talked about this earlier in the afternoon. Rene, do you need help working on this work item, reviewing, both?

**Comment by Rene2mt** [comment]
This is ready for review.  We also have issue [#90 ](https://github.com/GSA/automate.fedramp.gov/issues/90) but I'm thinking I work that one separately after this one (so the changes for that issue are easier to track).

**Review by aj-stein-gsa — COMMENTED** [review]
Sorry I have not updated sooner. It will take me some time to review this thoroughly, but I will come back to it tomorrow or Friday if I can.

It may be more advantageous, @Rene2mt, to set up a quick 15-25 min meeting and walk us through the PR and have some people on the call review along with you and then merge maybe? 🤷

**Comment by Rene2mt** [comment]
## Additional Details

Prior to this change, the FedRAMP High, Moderate, Low, and LI-SaaS profiles directly imported the NIST catalog.  This is illustrated below.

```mermaid
flowchart TD
    A[FedRAMP H/M/L/LI-SaaS R5 Profile] -->|imports| B[NIST SP800 53 R5 Catalog]
```

This PR refactors  the High, Moderate, Low, and LI-SaaS profiles by centralizing the majority of common alterations (e.g., response points, addition of parts for "FedRAMP Additional Guidance and Requirements", etc.) into the "tailoring" profile.  This is illustrated below.

```mermaid
flowchart TD
    A[FedRAMP H/M/L/LI-SaaS R5 Profile] -->|imports| B[FedRAMP Tailoring Profile]
    B -->|imports| C[NIST SP800 53 R5 Catalog]  
```

## Tips for PR Reviewers

Given the large number of changes (e.g., there are thousands of "response-point" `prop` assemblies moved from the H/M/L/LI-SaaS profiles to the "tailoring" profile"), reviewing the changes (e.g. diff) in GitHub may not work as well.  Here are some review tips:
- Review the PR in VSCode using the "GitHub pull requests" extension, that way you can see all the changes in the entire file
- Generate resolved profile catalogs from both the original profiles and the refactored profiles in this PR, and compare the resulting resolved profile catalogs.

File: src/content/rev5/baselines/xml/FedRAMP_rev5_catalog_tailoring_profile.xml [hunk 1]
```diff
@@ -0,0 +1,10360 @@
...
+			</add>
+			<add position="starting" by-id="ac-2.2">
+				<prop ns="https://fedramp.gov/ns/oscal" name="CORE" value="true"/>
+			</add>
+		</alter>
+		<alter control-id="ac-2.3">
+			<add position="ending" by-id="ac-2.3">
+				<part id="ac-2.3_fr" name="item" ns="https://fedramp.gov/ns/oscal">
+					<title>AC-2 (3) Additional FedRAMP Requirements and Guidance</title>
+					<part id="ac-2.3_fr_smt.1" name="item" ns="https://fedramp.gov/ns/oscal">
```
Comment (aj-stein-gsa) [inline src/content/rev5/baselines/xml/FedRAMP_rev5_catalog_tailoring_profile.xml:532]: Re the comment above in [](), an item with an item is allowed without a `@ns` namespace flag.

```suggestion
					<part id="ac-2.3_fr_smt.1" name="item">
```

There are a sequence of these controls with this pattern for item insertions like this, so this comment can track all of those fixes and tuning them in oXygen as you may.

**Review by aj-stein-gsa — CHANGES_REQUESTED** [review]

**Comment by brian-ruf** [comment]
@Rene2mt this is excellent wok so far. Per our discussion today, I'd suggest the following:
- descope LI-SaaS for now and leave the current Rev 5 version as-is.
- resolve the identifiers where FedRAMP additional requirements and guidance is inserted, to ensure it is inserted at the bottom of the control's `part[@name='statement']` rather than at the end of the bottom of the whole control. This should involve appending "_smt" to each of those identifiers.
- With the identifier correction above, remove the @ns values on all parts. 

Also, consider further reducing the scope of this PR to just one baseline. Produce a resolved profile catalog from the resulting stack (ie  FR_High -> FR Tailoring -> NIST Catalog) and compare to existing resolved profile catalogs to ensure they are equivalent from a context perspective.  Then create a separate issue and PR for the Moderate and Low baselines.

**Comment by aj-stein-gsa** [comment]
Re conversation and summary in https://github.com/GSA/fedramp-automation/issues/558#issuecomment-2452615652, we can wrap up some of the PR feedback with the given issues on the PR branch, but we will close this and re-open smaller PRs over time when this work comes back in scope for a sprint.

**Pull request closed without merging**